tests/unit/Owncloud/OcisPhpSdk/UserTest.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+namespace unit\Owncloud\OcisPhpSdk;
+
+use OpenAPI\Client\Model\User;
+use Owncloud\OcisPhpSdk\Exception\InvalidResponseException;
+use Owncloud\OcisPhpSdk\User as SdkUser;
+use PHPUnit\Framework\TestCase;
+
+class UserTest extends TestCase
+{
+    /**
+     * @return array<int, array<int, array<string, string>>>
+     */
+    public function validUserData(): array
+    {
+        return [
+            [[
+                "id" => "id",
+                "display_name" => "displayname",
+                "mail" => "mail@mail.com",
+                "on_premises_sam_account_name" => "sd",
+            ]],
+        ];
+    }
+
+    /**
+     * @dataProvider validUserData
+     * @param array<string,string|array<int,string>> $data
+     * @return void
+     */
+    public function testListGroup(array $data)
+    {
+        $libUser = new User($data);
+        $user = new SdkUser($libUser);
+        $this->assertInstanceOf(SdkUser::class, $user);
+        $this->assertEquals($data["id"], $user->getId());
+        $this->assertEquals($data["display_name"], $user->getDisplayName());
+        $this->assertSame($data["mail"], $user->getMail());
+        $this->assertSame($data["on_premises_sam_account_name"], $user->getOnPremisesSamAccountName());
+    }
+    /**
+     * @return array<int, array<int, string|null>>
+     */
+    public function invalidUserId(): array
+    {
+        return [[""], [null]];
+    }
+
+    /**
+     * @dataProvider invalidUserId
+     * @param string|null $data
+     * @return void
+     */
+    public function testInvalidId($data)
```

Comment:
all those invalid data tests can easily be combined into one, as you have done for groups